lib/ssz.ex

```diff
@@ -139,4 +154,305 @@ defmodule Ssz do
 
   @spec decode_u256(binary) :: non_neg_integer
   def decode_u256(num), do: :binary.decode_unsigned(num, :little)
+
+  ##### elixir native ssz
+  @bytes_per_length_offset 4
+  @bytes_per_chunk 32
+  @bits_per_byte 8
+
+  ### ENCODE ######
+  def serialize(%struct{} = map) when is_struct(map) do
+    schema = struct.schema()
+
+    values =
+      schema
+      |> Enum.map(fn schema ->
+        key = Enum.at(Map.keys(schema), 0)
+        metadata = Map.get(schema, key)
+        value = Map.get(map, key)
+        Map.merge(metadata, %{value: value})
+      end)
+
+    serialize(values)
+  end
+
+  def serialize(%{type: :struct, value: map, schema: schema}) do
+    serialize(map)
+  end
+
+  def serialize(%{type: :list, value: elements, schema: schema, max_size: max_size}) do
+    elements =
+      elements
+      |> Stream.map(fn value ->
+        Map.merge(schema, %{value: value})
+      end)
+      |> Enum.to_list()
+
+    serialize_list(elements)
+  end
+
+  def serialize(value) when is_list(value), do: serialize_list(value)
+
+  def serialize(%{type: :uint, size: size, value: value}) when is_integer(value),
+    do: serialize_uint(value, size)
```

Comment:
This was left from the previous iteration, right? The schema shouldn't have the value embedded in it.